### PR TITLE
Remove canSetPolicy, canUpdatePolicy, and canRemovePolicy

### PR DESCRIPTION
**Pull request by dakrone**
Since we now store a pre-compiled list of steps for an index's phase in the
`PolicyStepsRegistry`, we no longer need to worry about updating policies as any
updates won't affect the current phase, and will only be picked up on phase
transitions.

This also removes the tests that test these methods

Relates to #29823

### CODE REVIEW

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java [hunk 1]
```diff
@@ -353,106 +346,24 @@ public static ClusterState setPolicyForIndexes(final String newPolicyName, final
...
-            if (policyName.equals(currentPolicyName)) {
-                StepKey currentStepKey = IndexLifecycleRunner.getCurrentStepKey(idxSettings);
-                if (canSetPolicy(currentStepKey, currentPolicy, newPolicy) == false) {
-                    return false;
-                }
+        Settings.Builder newSettings = Settings.builder().put(idxSettings);
+        if (currentStepKey != null) {
+            // Check if current step exists in new policy and if not move to
+            // next available step
+            StepKey nextValidStepKey = newPolicy.getNextValidStep(currentStepKey);
```
Comment (colings86) [inline x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java]: I thinks the getNextValidStep() stuff can go to no?

Comment (dakrone) [reply x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java]: Can I do that as a followup? I think this change ended up larger than I hoped due to having to change a lot of tests

Comment (colings86) [reply x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java]: Sure.

**Review by colings86 — APPROVED** [review]
LGTM

**Pull request merged**